skaha/src/main/java/org/opencadc/skaha/K8SUtil.java

```diff
@@ -68,16 +68,24 @@
 package org.opencadc.skaha;
 
 import java.util.List;
+import java.util.Objects;
 import org.apache.log4j.Logger;
 
 public class K8SUtil {
+    private static final String VNC_URL_TEMPLATE =
```

Comment:
Just a thought,

- Maybe we should use a URL Building library to check for malformed urls, e.g. `okhttp3`
```Java
import okhttp3.HttpUrl;

HttpUrl url = new HttpUrl.Builder()
              .scheme("https")
              .host("www.example.com")
              .addPathSegment("path")
              .addQueryParameter("key", "value")
              .build();
System.out.println(url.toString());
```
- All the URI/URL work should probably reside in a separate component file, e.g. `URLBuilder`, rather than K8SUtil.